pkg/trace/api/api.go

```diff
@@ -456,6 +456,14 @@ func (r *HTTPReceiver) handleStats(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	firstService := func(cs *pb.ClientStatsPayload) string {
```

Comment:
curious why this is a small helper function here?

Reply:
Good question! At first, I wrote these 2 helpers as separate functions - unfortunately there is no straightforward way to write a generic one that covers both stats and traces. `handleStats` is relatively simple but `handleTraces` is quite long and does too many things, that's why I wanted the logic to be scoped in a helper even if it's used once.

Then, I thought it'd be a good idea to "protect" the package from relying on these helpers to get the service name for critical features in the future, that's why I encapsulated them as nested functions.